classes/Travel.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+require_once __DIR__ . "/MapLocation.php";
+
+class Travel {
+
+    const TRAVEL_DELAY_MOVEMENT = 400; // milliseconds
+    const TRAVEL_DELAY_PVP = 1; // seconds
+    const TRAVEL_DELAY_AI = 1; // seconds
+    const TRAVEL_DELAY_DEATH = 15; // seconds
+    const HOME_VILLAGE_COLOR = 'FFEF30';
+    const PLAYER_ICON = '/images/ninja_head.png';
+    const DEFAULT_MAP_ID = 1;
+
+    public function __construct() {
+    }
+
+    public static function getNewMovementValues(string $direction, TravelCoords $current): TravelCoords {
+        $new_x = $current->x;
+        $new_y = $current->y;
+        switch($direction) {
+            case 'north':
+                $new_y -= 1;
+                break;
+            case 'south':
+                $new_y += 1;
+                break;
+            case 'west':
+                $new_x -= 1;
+                break;
+            case 'east':
+                $new_x += 1;
+                break;
+            case 'northwest':
+                $new_x -= 1;
+                $new_y -= 1;
+                break;
+            case 'northeast':
+                $new_x += 1;
+                $new_y -= 1;
+                break;
+            case 'southwest':
+                $new_x -= 1;
+                $new_y += 1;
+                break;
+            case 'southeast':
+                $new_x += 1;
+                $new_y += 1;
+                break;
+        }
+        return new TravelCoords($new_x . '.' . $new_y . '.' . $current->z);
+    }
+
+    public static function checkPVPDelay(int $last_pvp): int {
+        $diff = time() - $last_pvp;
+        return self::TRAVEL_DELAY_PVP - $diff;
+    }
+
+    public static function checkAIDelay(int $last_ai): int {
+        $diff = time() - $last_ai;
+        return self::TRAVEL_DELAY_AI - $diff;
+    }
```

Comment:
I think these two need to use milliseconds like the movement delay does, so we have full control over how long this window is.